src/backend/EdFi.DataManagementService.Backend.Postgresql/Operation/SqlAction.cs

```diff
@@ -161,6 +172,55 @@ NpgsqlTransaction transaction
         return await upsertDocumentCmd.ExecuteNonQueryAsync();
     }
 
+    public async Task<UpdateDocumentValidationResult> UpdateDocumentValidation(
+        DocumentUuid documentUuid,
+        PartitionKey documentPartitionKey,
+        ReferentialId referentialId,
+        PartitionKey referentialPartitionKey,
+        NpgsqlConnection connection,
+        NpgsqlTransaction transaction)
+    {
+        await using NpgsqlCommand validationCommand = new(
+            @"SELECT DocumentUuid, ReferentialId
+                FROM public.documents d
+                LEFT JOIN public.aliases a ON
+                    a.DocumentId = d.Id
+                    AND a.DocumentPartitionKey = d.DocumentPartitionKey
+                    AND a.ReferentialId = $1 and a.ReferentialPartitionKey = $2
+                WHERE d.DocumentUuid = $3 AND d.DocumentPartitionKey = $4",
```

Comment:
Upon reflection, I'm less sure about whether this query does a table/partition scan. Can you run an explain on it and check?

Reply:
Looks good:

Reply:
Great, thanks!